fbw-a380x/src/systems/instruments/src/PFD/FMA.tsx

```diff
@@ -69,6 +72,38 @@ export class FMA extends DisplayComponent<{ bus: EventBus; isAttExcessive: Subsc
 
   private AB3Message = Subject.create(false);
 
+  private isUnrestrictedClimbDescent: number = 0;
+
+  private readonly sub = this.props.bus.getSubscriber<PFDSimvars & Arinc429Values & SimplaneValues>();
+
+  private readonly selectedFpa = ConsumerSubject.create(this.sub.on('selectedFpa'), 0);
+
+  private readonly selectedVs = ConsumerSubject.create(this.sub.on('selectedVs'), 0);
+
+  private readonly selectedAltitude = ConsumerSubject.create(this.sub.on('selectedAltitude'), 0);
+
+  private readonly altitude = Arinc429ConsumerSubject.create(this.sub.on('altitudeAr'));
+
+  private readonly activeVerticalModeConsumer = ConsumerSubject.create(this.sub.on('activeVerticalMode'), 0);
+
+  private readonly UnrestrictedClimbDescent = MappedSubject.create(
+    ([selectedFpa, selectedVs, altitude, selectedAltitude, activeVerticalMode]) => {
+      if (activeVerticalMode === VerticalMode.FPA || activeVerticalMode === VerticalMode.VS) {
+        if ((selectedFpa > 0 || selectedVs > 0) && selectedAltitude < altitude.value) {
+          return 1;
+        } else if ((selectedFpa < 0 || selectedVs < 0) && selectedAltitude > altitude.value) {
+          return 2;
+        }
+      }
+      return 0;
+    },
+    this.selectedFpa,
+    this.selectedVs,
+    this.altitude,
+    this.selectedAltitude,
+    this.activeVerticalModeConsumer,
+  ) as Subscribable<number>;
```

Comment:
```suggestion
  );
```

correct type should be inferred automatically, if not something in your MappedSubject function is wrong

Reply:
Yup, it caused some trouble before for some reason but now it's okay. Removed it 👍